tools/SourceKit/tools/sourcekitd/bin/InProc/sourcekitdInProc.cpp

```diff
@@ -102,16 +102,17 @@ void sourcekitd::set_interrupted_connection_handler(
 //===----------------------------------------------------------------------===//
 
 sourcekitd_response_t sourcekitd_send_request_sync(sourcekitd_object_t req) {
-  dispatch_semaphore_t sema = dispatch_semaphore_create(0);
+  sourcekitd_response_t ReturnedResp = nullptr;
+  std::condition_variable handleRequestCondition;
 
-  sourcekitd_response_t ReturnedResp;
   sourcekitd::handleRequest(req, [&](sourcekitd_response_t resp) {
     ReturnedResp = resp;
```

Comment:
The typical pattern is to use the same mutex to protect writing to the condition being waited on.